explore_RNAseq_only_generic_genes/1c_original_vs_simulated_counts.py

```diff
@@ -0,0 +1,227 @@
+            fontsize=16,
+            fontname="Verdana",
+        )
+    elif by_sample_or_gene == "gene":
+        for i in range(num_simulated):
+            # Get simulated filename
+            simulated_filename = os.path.join(
+                simulated_dir, f"selected_simulated_data_{project_id}_{i}_processed.txt"
+            )
+
+            # Read simulated experiment
+            simulated = pd.read_csv(simulated_filename, sep="\t", index_col=0, header=0)
+
+            # Create dataframe with columns
+            # Gene id | total counts across samples original (template)| total counts across samples after VAE (simulated)
+            template_gene_counts = template.sum()
+            simulated_gene_counts = simulated.sum()
+            # template_gene_counts = template.iloc[1]
+            # simulated_gene_counts = simulated.iloc[1]
+
+            gene_counts = template_gene_counts.to_frame("Original counts").merge(
+                simulated_gene_counts.to_frame("Simulated counts"),
+                left_index=True,
+                right_index=True,
+            )
+
+            # Plot per sample
+            g = sns.scatterplot(
+                data=gene_counts,
+                x="Simulated counts",
+                y="Original counts",
+                alpha=0.3,
+                ax=axes[i],
+            )
+            # axes[i].hexbin(x=gene_counts["Simulated counts"],
+            #               y=gene_counts["Original counts"],
+            #               gridsize=100
+            #              )
+            # axes[i].set_ylim([0, 0.1e7])
+            # axes[i].set_xlim([0, 0.1e7])
+
+            min_ = gene_counts.min().min()
+            max_ = gene_counts.max().max()
+            g.plot([min_, max_], [min_, max_], "k--")
+            g.plot([0, 0.1e7], [0, 0.1e7], "k--")
+            axes[i].set_ylabel("")
+            axes[i].set_xlabel("")
+
+        fig.text(
+            0.5,
+            0.0,
+            "Simulated counts (after VAE)",
+            ha="center",
+            fontsize=14,
+            fontname="Verdana",
+        )
+        fig.text(
+            0.08,
+            0.5,
+            "Actual counts",
+            va="center",
+            rotation="vertical",
+            fontsize=14,
+            fontname="Verdana",
+        )
+        fig.suptitle(
+            "Actual vs simulated counts per gene",
+            fontsize=16,
+            fontname="Verdana",
+        )
+
+
+# Compare per sample
+scatter_plot_original_vs_simulated(
+    ncols=5,
+    nrows=5,
+    fig_width=20,
+    fig_height=20,
+    num_simulated=25,
+    by_sample_or_gene="sample",
+)
+
+# **Observations:**
+# * For about half of the cases, here is a horizontal trend that indicates that the variance in the actual total counts is lower compared to the simulated total counts. In other words the sequencing coverage of the actual experiment is consistent while the sequencing coverage of the simulated samples is variable.
```

Comment:
So for the per-sample plots, what you would normally expect to see is a roughly linear relationship (i.e. the simulated counts shouldn't be exactly the same as the actual counts, but they should roughly have the same trend; higher actual counts should generally lead to higher simulated counts)?

If so, I agree, that doesn't seem to be the case in most of these experiments.

I wonder what these plots look like for randomly sampled genes (i.e. genes that aren't lowly expressed in the RNA-seq data, or even genes that aren't generic) - have you thought at all about looking at that?

Reply:
So I would expect that the sequencing coverage of samples within a simulated experiment to be very tight (so not much spread horizontally. I would also expected that the sequencing coverage of the simulated experiment to be similar to the actual experiment, so the samples should cluster along the diagonal.

Reply:
I'm still thinking through what it would mean to plot different subsets. So you can look at total counts across the RNA-seq/array generic genes between actual vs simulated. You can also look at total counts across the RNA-seq only generic genes between actual vs simulated. But I'm not sure how to interpret the results here. 

So in general, it looks like the RNA-seq/array generic genes have higher counts in the actual experiment while RNA-seq only generic genes have lower counts in the actual experiment. You still see difference in the total counts between actual vs simulation for both sets of genes.

Reply:
In terms of correction, I think that plan right now is to re-scale the simulated experiment counts to be match the actual counts. So the relative gene counts within the simulated vs actual experiment can reflect different experiments. But the total counts will be the same between actual and simulated.